src/wp-includes/media.php

```diff
@@ -1843,6 +1845,11 @@ function wp_filter_content_tags( $content, $context = null ) {
 				$filtered_image = wp_img_tag_add_loading_attr( $filtered_image, $context );
 			}
 
+			// Add 'decoding=async' attribute unless a 'decoding' attribute is already present.
+			if ( ! str_contains( $filtered_image, ' decoding=' ) ) {
+				$filtered_image = wp_img_tag_add_decoding_async_attr( $filtered_image, $context );
+			}
```

Comment:
I'd suggest moving the conditional check to the function. See my notes here https://github.com/WordPress/wordpress-develop/pull/2409/files#r837622678.

```suggestion
			$filtered_image = wp_img_tag_add_decoding_async_attr( $filtered_image, $context );
```

Reply:
oh, ok. my only hesitation is that the other functions in this file that make similar changes to img tags all do the check for existing attribute in the outside function (here) vs in the function. So for consistency thinking it should stay the way it is, what do you think?

Reply:
I think the guard clause is needed in the new `wp_img_tag_add_decoding_async_attr()` function rather than where consumed. Why? It's a new global function and could be invoked from anywhere. Adding the guard clause within the function avoids unnecessary processing work and guards against inadvertently adding the attribute more than once.

What about consistency in this area of the code? IMO the same logic applies to the other attribute instances, i.e. guarding within the function rather than where consumed. But that would be out of scope for this PR. 

Do you think a follow-up PR to modify all of them (including this one) is a better approach? Or introduce the pattern now in this PR and then update the other instances in a separate PR?

Reply:
yea, a follow up PR is probably fine. good point about this being a globally accessible function, I'll move the check into the function.